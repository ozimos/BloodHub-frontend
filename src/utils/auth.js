const tokenKey = 'token';

const saveUserToken = token => window.localStorage.setItem( tokenKey, token );
const getUserToken = () => window.localStorage.getItem( tokenKey );
const deleteUserToken = () => window.localStorage.removeItem( tokenKey );

export default {
  saveUserToken,
  getUserToken,
  deleteUserToken,
};
