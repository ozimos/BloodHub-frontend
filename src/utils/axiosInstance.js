import axios from 'axios';
import authUtils from './auth';

const instance = axios.create( {
  baseURL: process.env.REACT_APP_API_URL || 'https://secret-island-30539.herokuapp.com/',
} );

instance.interceptors.request.use( ( config ) => {
  const configInstance = { ...config };
  configInstance.headers.Authorization = authUtils.getUserToken();
  return configInstance;
} );

export default instance;
