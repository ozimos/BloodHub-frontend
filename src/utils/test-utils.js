import React from "react";
import { ThemeProvider } from "@material-ui/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import { MockedProvider } from "@apollo/react-testing";
import { render } from "@testing-library/react";
import { REGISTER_USER, LOG_IN_USER } from "apolloUtils/requests";
import { resolvers as defaultResolvers } from "apolloUtils/resolvers";

export const defaultMocks = [
  {
    request: {
      mutation: REGISTER_USER,
      variables: {
        data: {
          firstName: "Buck",
          lastName: "Buck",
          email: "hail@hydra.com",
          password: "c@ptainAm3r1c@",
          phone: "66699969",
          donor: {
            create: { bloodGroup: "A_negative" }
          }
        }
      }
    },
    result: {
      data: {
        [REGISTER_USER]: {
          isLoggedIn: true,
          user: { firstName: "Buck", lastName: "Buck", isDonor: true },
          token: "randomtoken"
        }
      }
    }
  },
  {
    request: {
      mutation: LOG_IN_USER,
      variables: {
        data: {
          email: "hail@hydra.com",
          password: "c@ptainAm3r1c@"
        }
      }
    },
    result: {
      data: {
        [LOG_IN_USER]: {
          isLoggedIn: true,
          user: { firstName: "Buck", lastName: "Buck", isDonor: true },
          token: "randomtoken"
        }
      }
    }
  }
];

export const configureProviderWrapper = ({
  mocks = [],
  theme = {},
  resolvers = defaultResolvers,
  ...rest
} = {}) => ({ children }) => (
  <MockedProvider
    mocks={mocks}
    resolvers={resolvers}
    addTypename={false}
    {...rest}
  >
    <ThemeProvider theme={createMuiTheme(theme)}>{children}</ThemeProvider>
  </MockedProvider>
);

const customRender = (ui, wrapper = configureProviderWrapper(), options) =>
  render(ui, { wrapper, ...options });

export * from "@testing-library/react";

export { customRender as render };
