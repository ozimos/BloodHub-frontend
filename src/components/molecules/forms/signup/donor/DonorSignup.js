import React from "react";
import { object } from "yup";
import AuthForm from "../../AuthForm";
import BaseSignupForm, {baseInitialValues, baseValidationFields} from "../BaseSignupForm";
import DonorFields, {donorInitialValues, donorValidationFields} from "./DonorFields";
import { signup } from "../../../../../actions/auth";

export const initialValues = {...baseInitialValues, ...donorInitialValues};

export const validationSchema = object({...baseValidationFields, ...donorValidationFields});

export default () => (
  <AuthForm
    action={signup}
    redirectPath="/dashboard"
    validationSchema={validationSchema}
    initialValues={initialValues}
  >
  <BaseSignupForm component={DonorFields} />
</AuthForm>
);
