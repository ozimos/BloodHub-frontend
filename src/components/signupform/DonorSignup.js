import React from "react";
import { object } from "yup";
import PartialForm, {baseInitialValues, baseValidationFields} from "./signupform/PartialForm";
import DonorFields, {donorInitialValues, donorValidationFields} from "./signupform/DonorFields";
import { signup } from "../../actions/auth";

export const initialValues = {...baseInitialValues, ...donorInitialValues};

export const validationSchema = object({...baseValidationFields, ...donorValidationFields});

export default () => (
  <PartialForm
    action={signup}
    component={DonorFields}
    redirectPath="/dashboard"
    validationSchema={validationSchema}
    initialValues={initialValues}
  />
);
