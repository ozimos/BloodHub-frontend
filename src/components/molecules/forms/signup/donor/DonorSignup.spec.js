import React from "react";
import { getAllByLabelText } from "@testing-library/dom";
import { MemoryRouter } from "react-router-dom";
import { toMatchDiffSnapshot } from "snapshot-diff";
import { render, fireEvent, act } from "utils/test-utils";
import DonorSignup from "./DonorSignup";

expect.extend({ toMatchDiffSnapshot });

const labels = [
  "First Name",
  "Last Name",
  "Email Address",
  "Password",
  "Verify Password",
  "Blood Group",
  "Street Address",
  "Local Government",
  "State"
];

const Wrapper = () => (
  <MemoryRouter>
    <DonorSignup />
  </MemoryRouter>
);

it("renders all required input fields with error labels", async () => {
  const { container, getByText, asFragment } = render(<Wrapper />);
  const firstRender = asFragment();
  const submitButton = getByText(/Proceed/i);

  const inputFields = getAllByLabelText(
    container,
    content => labels.some(label => content.startsWith(label)),
    {
      selector: "input",
      exact: false
    }
  );
  expect(inputFields).toHaveLength(labels.length);

  await act(async () => {
    inputFields.forEach(field => fireEvent.blur(field, {}));
    fireEvent.click(submitButton, {});
  });

  expect(firstRender).toMatchDiffSnapshot(asFragment(), {
    contextLines: 0,
    stablePatchmarks: true
  });
});
