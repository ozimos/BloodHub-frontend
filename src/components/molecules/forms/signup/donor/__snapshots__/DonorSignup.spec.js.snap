// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all required input fields with error labels 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
-             data-shrink="true"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
+             data-shrink="false"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="firstName-helper-text"
+               aria-invalid="true"
@@ --- --- @@
-                 style="width: 0px;"
+                 style="width: 0.01px;"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="firstName-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="lastName-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="lastName-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="email-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="email-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
+               aria-describedby="bloodGroup-helper-text"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="bloodGroup-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="street-address-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="street-address-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="lg-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="lg-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="state-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="state-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="password-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="password-helper-text"
+           >
+             Required
+           </p>
@@ --- --- @@
-             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
+             class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-error Mui-error Mui-required Mui-required"
@@ --- --- @@
-               class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
+               class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error Mui-error"
@@ --- --- @@
-             class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
+             class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
@@ --- --- @@
-               aria-invalid="false"
+               aria-describedby="verifyPassword-helper-text"
+               aria-invalid="true"
@@ --- --- @@
+           <p
+             class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error Mui-required"
+             id="verifyPassword-helper-text"
+           >
+             Required
+           </p>
`;
